app/src/main/java/com/arctouch/codechallenge/web/api/TmdbApi.kt

```diff
@@ -0,0 +1,54 @@
+package com.arctouch.codechallenge.web.api
+
+import com.arctouch.codechallenge.model.GenreResponse
+import com.arctouch.codechallenge.model.UpcomingMoviesResponse
+import io.reactivex.Observable
+import okhttp3.OkHttpClient
+import retrofit2.Retrofit
+import retrofit2.adapter.rxjava2.RxJava2CallAdapterFactory
+import retrofit2.converter.moshi.MoshiConverterFactory
+import retrofit2.http.GET
+import retrofit2.http.Query
+
+interface TmdbApi {
+
+    companion object {
+        private const val URL = "https://api.themoviedb.org/3/"
+        const val API_KEY = "1f54bd990f1cdfb230adb312546d765d"
+        const val DEFAULT_LANGUAGE = "pt-BR"
+        const val DEFAULT_REGION = ""
+
+        fun create(): TmdbApi {
+            val retrofit = Retrofit.Builder()
+                    .baseUrl(URL)
+                    .client(OkHttpClient.Builder().build())
+                    .addConverterFactory(MoshiConverterFactory.create())
+                    .addCallAdapterFactory(RxJava2CallAdapterFactory.create())
+                    .build()
+            return retrofit.create(TmdbApi::class.java)
+        }
+    }
+
+    @GET("genre/movie/list")
+    fun genres(
+            @Query("api_key") apiKey: String = API_KEY,
```

Comment:
Do you have heard about the `OkHttp` interceptors? Here and below, notice that you're using `api_key`, `language` and `region` almost on each request call. To avoid passing then, you can create an Request Interceptor and add them just before each call. More details on how to do this is here: https://github.com/square/okhttp/wiki/Interceptors

Reply:
Yeah!!, I already used the interceptors for this purpose. How can I forgot that?!..rsrsr
Nice